test/dump.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *  
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {assert} from 'chai';
+import * as ion from '../src/IonTests';
+import {dom, exampleIonValuesWhere, exampleJsValuesWhere, IonTypes} from "../src/IonTests";
+
+function roundTripTests(...values: any[]) {
+    let expectedValues: dom.Value[] = values.map((jsValue) => dom.Value.from(jsValue));
+    it(`dumpBinary(${values})`, () => roundTripTest(ion.dumpBinary, values, expectedValues));
+    it(`dumpText(${values})`,() => roundTripTest(ion.dumpText, values, expectedValues));
+    it(`dumpPrettyText(${values})`, () => roundTripTest(ion.dumpPrettyText, values, expectedValues));
+}
+
+// Dumps the provided values to Ion using the specified dump*() function, uses loadAll() to read the values
+// back in, then compares the input values to the loaded values.
+function roundTripTest(dumpFn: (...values: any[]) => any,
+                       inputValues: any[],
+                       expectedValues: dom.Value[]) {
+    let actualValues: dom.Value[] = dom.loadAll(dumpFn(inputValues));
+    let testComparisons: [dom.Value, dom.Value][] = actualValues.map(
+        (actualValue, index) => [actualValue, expectedValues[index]]
+    );
```

Comment:
does `testComparisons` add value over having the `for` loop below iterate over `actualValues`?

Reply:
I need access to the index of `actualValue`, which I can't get from
```typescript
for (let actualValue of actualValues) {
  let expectedValue = expectedValues[???];
  // Compare them
}
```

Zipping the two arrays together in advance makes for a nice, self-explanatory loop header:
```typescript
for (let [actualValue, expectedValue] of testComparisons) {}
```

Unfortunately, Javascript doesn't provide a `zip` method out of the box so you end up seeing how the sausage gets made.

I've converted this to a `forEach` in the hopes that that's more palatable:
```typescript
expectedValues.forEach((expectedValue, index) => {
   let actualValue = actualValues[index];
   // Compare them
}
```